File: csrc/list-tithis.py
import sys
import re
with open("tithi-input.txt") as fp:
	line = fp.readline().strip()
	while line:
		#try:
		if True:
			print (line)
			#r = re.compile('^\d+\/\d+\/\d+ \d+:\d+ [A|P]M\s+\d+\.\S+\s*')
			r = re.compile('^\d+\/\d+\/\d+ \d+:\d.*')
			line = fp.readline().strip()
			m = r.match(line)
			print (m)
		#except:
		else:
			print ("Error with line ", line)
			sys.exit()
		

File: csrc/list-tithis.pl
open(INPUT, "<tithi-input.txt") or die "Couldn't open file tithi-input.txt, $!";

while($line = <INPUT>) {
	$line =~ s/^\s+|\s+$//g;
	#print "$line\n";
	if ($line =~ /^(\d+)\/(\d+)\/(\d+).*\s(\d+):(\d+)\s([A|P])M\s+\d+\.\s+(\S+)\s.*$/) {
		$hr = $4;
		if ($6 eq "P" && $4 < 12) {
			$hr += 12;
		} elsif ($6 eq "A" && $4 == 12) {
			$hr = "0";
		}
		$yr = 2000 + $3;
		print "$2-$1-$yr,$hr:$5,$7\n";
	}
}
